### DISCUSSION TITLE
ArrayFunctions like sort and sortby

### DISCUSSION THREAD

**Discussion by jr212** [PHPOffice/PhpSpreadsheet#3864] [Q&A]
Just tested version 2.0-Development hoping that I can use the new sort functions (https://support.microsoft.com/en-au/office/sortby-function-cd2d7a62-1b93-435c-b561-d6a35134f28f)
Unfortunately  without luck. Meaning the function starts but must fill multiple rows and columns and only 1 cell is filled and a @ is in front of the formula. How can I correctly enter the formula?

Also How can I use only and nothing else that the equal sign in a cell? In Excel add a ‘ in front of the formula does the trick but here it’s Error 😊

**Comment by oleibman** [comment]
If you can test against PR #3962, please do so and let me know if it addresses your problem.

As for equals sign in a cell, one way is:
```php
$sheet->setCellValueExplicit('A1','=', DataType::TYPE_STRING);
```

**Comment by jr212** [comment]
Hi

 

I’ve tested both.

*	On the =sign I get an error: Fatal error: Uncaught Error: Class "DataType" not found
*	For the @ nothing changed

 

I composed the library with composer require phpoffice/phpspreadsheet:dev-master

Must I have used other options?

 

Jan

 

Van: oleibman ***@***.***> 
Verzonden: maandag 15 april 2024 5:01
Aan: PHPOffice/PhpSpreadsheet ***@***.***>
CC: jr212 ***@***.***>; Author ***@***.***>
Onderwerp: Re: [PHPOffice/PhpSpreadsheet] ArrayFunctions like sort and sortby (Discussion #3864)

 

If you can test against PR #3962 <https://github.com/PHPOffice/PhpSpreadsheet/pull/3962> , please do so and let me know if it addresses your problem.

As for equals sign in a cell, one way is:

$sheet->setCellValueExplicit('A1','=', DataType::TYPE_STRING);

**Reply by oleibman** [reply]
To avoid the DataType problem:
```php
use PhpOffice\PhpSpreadsheet\Cell\DataType;;
```

**Comment by jr212** [comment]
hi

the =-sign works fine. However not for multiple cells. It would be nice to
note that in the manual.
thanks.

Any news for the @-problem?

Jan

Op di 16 apr 2024 om 15:45 schreef oleibman ***@***.***>:

> To avoid the DataType problem:
>
> use PhpOffice\PhpSpreadsheet\Cell\DataType;;
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PHPOffice/PhpSpreadsheet/discussions/3864#discussioncomment-9130457>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKXCIPLAFV2TRXOMPECUYX3Y5UTPFAVCNFSM6AAAAABBWI2QCGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCMZQGQ2TO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Met vriendelijke groeten
Jan
***@***.***

**Reply by oleibman** [reply]
No news about at-sign. As I said, testing with the code from PR 3962 seems to work adequately, albeit not perfectly.

Also, it seems that `$sheet->getCell('A1')->setValue('=');` works without having to use `setValueExplicit`. If you have a case where it doesn't please suppply more information. I am not sure what you mean by 'not for multiple cells'.

**Comment by jr212** [comment]
Multiplecell like range a1:b2

Op wo 17 apr. 2024 04:14 schreef oleibman ***@***.***>:

**Comment by oleibman** [comment]
SORT and SORTBY have been implemented for some time now.
